### PR TITLE
[opt](stats) Scale replica of stats table to 3 when it's possible

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

So that we could improve the availability of stats.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java [hunk 1]
```diff
@@ -75,6 +77,46 @@ public void run() {
...
+        modifyTblReplicaCount(database, StatisticConstants.STATISTIC_TBL_NAME);
+        modifyTblReplicaCount(database, StatisticConstants.HISTOGRAM_TBL_NAME);
+    }
+
+    public void modifyTblReplicaCount(Database database, String tblName) {
+        if (!(Config.min_replication_num_per_tablet < 3 && Config.max_replication_num_per_tablet >= 3)) {
+            return;
+        }
+        while (true) {
+            if (Env.getCurrentSystemInfo().aliveBECount() >= 3) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java:97]: check tables current replica num, if all tables' replica num is 3 then exit this thread

File: fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java [hunk 2]
```diff
@@ -984,4 +984,9 @@ public int getMinPipelineExecutorSize() {
         }
         return minPipelineExecutorSize;
     }
+
+    public long aliveBECount() {
+        Map<Long, Backend> copiedBackends = Maps.newHashMap(idToBackendRef);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java:989]: not need copy map when read this map

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.29 seconds
 stream load tsv:          538 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          30.1 seconds inserted 10000000 Rows, about 332K ops/s
 storage size: 17162608162 Bytes

**Comment by Kikyou1997** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java [hunk 3]
```diff
@@ -75,6 +77,52 @@ public void run() {
...
+    public void modifyTblReplicaCount(Database database, String tblName) {
+        if (!(Config.min_replication_num_per_tablet < StatisticConstants.STATISTIC_INTERNAL_TABLE_REPLICA_NUM
+                && Config.max_replication_num_per_tablet >= StatisticConstants.STATISTIC_INTERNAL_TABLE_REPLICA_NUM)) {
+            return;
+        }
+        while (true) {
+            if (Env.getCurrentSystemInfo().aliveBECount() >= StatisticConstants.STATISTIC_INTERNAL_TABLE_REPLICA_NUM) {
+                try {
+                    Map<String, String> props = new HashMap<>();
+                    props.put(PropertyAnalyzer.PROPERTIES_REPLICATION_ALLOCATION, "tag.location.default: 3");
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java:101]: "tag.location.default: 3" =>"tag.location.default: “+StatisticConstants.STATISTIC_INTERNAL_TABLE_REPLICA_NUM

File: fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java [hunk 4]
```diff
@@ -75,6 +77,52 @@ public void run() {
...
+                        Env.getCurrentEnv().modifyTableReplicaAllocation(database, (OlapTable) colStatsTbl, props);
+                    } finally {
+                        colStatsTbl.writeUnlock();
+                    }
+                    break;
+                } catch (Throwable t) {
+                    LOG.warn("Failed to scale replica of stats tbl:{} to 3", tblName, t);
+                }
+                try {
+                    Thread.sleep(5000);
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java:120]: this sleep should be moved into else block

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.76 seconds
 stream load tsv:          532 seconds loaded 74807831229 Bytes, about 134 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17163496950 Bytes

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**